File: java/Memory.java
package latihan2.java;

public class Memory extends Hardware {

    private int frequency;
    private int memorySize;
    private Boolean supportCuda;

    public Memory() {
    }

    public Memory(int frequency, int memorySize, Boolean supportCuda) {
        this.setFrequency(frequency);
        this.setMemorySize(memorySize);
        this.setSupportCuda(supportCuda);
    }

    public Boolean getSupportCuda() {
        return supportCuda;
    }

    public void setSupportCuda(Boolean supportCuda) {
        this.supportCuda = supportCuda;
    }

    public int getMemorySize() {
        return memorySize;
    }

    public void setMemorySize(int memorySize) {
        this.memorySize = memorySize;
    }

    public int getFrequency() {
        return frequency;
    }

    public void setFrequency(int frequency) {
        this.frequency = frequency;
    }

}

File: python/Memory.py
from Hardware import Hardware


class Memory(Hardware):

    def __init__(self, frequency=0, memorySize=0, supportCuda=0):
        self.__frequency = frequency
        self.__memorySize = memorySize
        self.__supportCuda = supportCuda

    def setFrequency(self, frequency):
        self.__frequency = frequency

    def setMemorySize(self, memorySize):
        self.__memorySize = memorySize

    def setSupportCuda(self, supportCuda):
        self.__supportCuda = supportCuda

    def getFrequency(self):
        return self.__frequency

    def getMemorySize(self):
        return self.__memorySize

    def getSupportCuda(self):
        return self.__supportCuda
